Remove the name 'Rubinius'

chrisseaton opened the pull request:
We don't want gems finding it through feature detection, and there's no distinction between our code and theirs now - let's mix it all in together.

Basically, `Rubinius` is now `Truffle::Support`, with the idea that things can move into `Truffle` in the future (not always simple).

eregon commented:
Please let me review before merging.
Might be good to simplify some things so we don't have deeply-nested namespaces which is verbose and cumbersome.

chrisseaton commented:
@eregon if you want I can try moving most of the modules and classes into just `Truffle`. A couple of them conflict which is why I didn't do it directly here.

eregon commented on doc/user/interop.md:
```diff
@@ -48,14 +48,14 @@ Call `size` on the object.
-Returns true only for instances of `String`, `Symbol`, `Rubinius::FFI::Pointer`
-and objects that respond to `unbox`.
+Returns true only for instances of `String`, `Symbol`,
+`Truffle::Support::FFI::Pointer` and objects that respond to `unbox`.
```
Can we simplify Truffle::Support::FFI to just Truffle::FFI? It's way too long.

chrisseaton replied:
Yes, as I've said we can move things into `Truffle`, I just haven't done it yet.

eregon commented on lib/patches/bundler/bundler/ruby_version.rb:
```diff
@@ -15,7 +15,7 @@ def self.system
                               ruby_version
                             when "rbx"
-                              Rubinius::VERSION.dup
+                              Truffle::Support::VERSION.dup
```
This should not be modified (it's upstream code)

chrisseaton replied:
Fixed.

eregon commented on lib/rubysl/socket/rubysl.rb:
```diff
@@ -27,11 +27,11 @@
   module Socket
     def self.bsd_support?
-      Rubinius.bsd? || Rubinius.darwin?
+      Truffle::Support.bsd? || Truffle::Support.darwin?
```
I think this would be nice as `Truffle::Platform.bsd?`

chrisseaton replied:
Yes, again, we can move things into `Truffle` I just haven't done it yet.

eregon commented on lib/rubysl/socket/socket.rb:
```diff
@@ -261,7 +261,7 @@ class << self
   if RubySL::Socket.unix_socket_support?
     def self.pack_sockaddr_un(file)
-      max_path_size =  Rubinius::Config['rbx.platform.sockaddr_un.sun_path.size'] - 1
+      max_path_size =  Truffle::Support::Config['rbx.platform.sockaddr_un.sun_path.size'] - 1
```
Can it be just `Truffle::Config` or `Truffle::NativeConfig` or `Truffle::Platform`? (I prefer `Truffle::Platform`)

chrisseaton replied:
You're asking the same question for all of these...

eregon commented on src/main/java/org/truffleruby/core/string/StringNodes.java:
```diff
@@ -517,8 +517,8 @@ public Object sliceObjectRange(VirtualFrame frame, DynamicObject string, Dynamic
             // TODO (nirvdrum 31-Mar-15) The begin and end values may return Fixnums beyond int boundaries and we should handle that -- Bignums are always errors.
-            final int coercedBegin = (int) snippetNode1.execute(frame, "Rubinius::Type.rb_num2int(v)", "v", Layouts.OBJECT_RANGE.getBegin(range));
-            final int coercedEnd = (int) snippetNode2.execute(frame, "Rubinius::Type.rb_num2int(v)", "v", Layouts.OBJECT_RANGE.getEnd(range));
+            final int coercedBegin = (int) snippetNode1.execute(frame, "Truffle::Support::Type.rb_num2int(v)", "v", Layouts.OBJECT_RANGE.getBegin(range));
```
Can it be just `Truffle::Type`/

chrisseaton replied:
You're asking the same question for all of these...

eregon replied:
Yes, I just wanted to give examples of the important ones we should simplify

eregon commented:
> A couple of them conflict which is why I didn't do it directly here.

Which ones conflict?
I think it would be best to avoid an extra namespace where possible `Truffle::Support::Type.object_class(object)` starts to look as verbose as Java code.
I think it should be done in this PR, so the diff is more straightforward and easier to review.

chrisseaton commented:
For example, in `type.rb`, `Encoding` would then refer to `Truffle::Encoding` rather than `::Encoding`. It's possible that this is the only issue and I can move the rest into just `Truffle`, but that's why I didn't immediately do it.

chrisseaton commented:
Shall I move the pain ones (`Type`, `FFI`, `Mirror` etc) into `Truffle` and stop when I find ones I can't move?

eregon commented:
I would rename Truffle::Encoding to Truffle::EncodingOperations like you did for Truffle::Fixnum. Yeah, I think having the non problematic ones directly under Truffle would make code nicer. Then we can think about the problematic ones and maybe simplify their namespace later as I think it's only a couple or so.

nirvdrum commented on src/main/java/org/truffleruby/language/RubyGuards.java:
```diff
@@ -197,7 +197,7 @@ public static boolean isRubyPointer(DynamicObject value) {
     }
 
-    public static boolean isRubiniusByteArray(DynamicObject value) {
+    public static boolean isSupportByteArray(DynamicObject value) {
```
As part of the rename, it'd be nice if this were renamed to `isTruffleByteArray`. Just something to note when you change the namespace again.

chrisseaton replied:
Will do.

nirvdrum commented on src/main/java/org/jcodings/transcode/TranscodingManager.java:
```diff
@@ -140,11 +140,11 @@ private static EConv create(byte[] sourceEncodingName, byte[] destinationEncodin
      * We figure out what flags JRuby additionally expects to be set and set them to satisfy EConv.
      */
-    private static int rubiniusToJRubyFlags(int flags) {
+    private static int toJRubyFlags(int flags) {
```
Unrelated to your change here, but we could call this `toJCodingsFlags`.

chrisseaton replied:
Will do.

nirvdrum commented on lib/patches/bundler/bundler/ruby_version.rb:
```diff
@@ -15,7 +15,7 @@ def self.system
                               ruby_version
                             when "rbx"
-                              Truffle::Support::VERSION.dup
+                              Truffle::VERSION.dup
```
To avoid confusion, you might want to just squash this into the commit that changed it and remove all history that this was ever changed incorrectly.

eregon replied:
@chrisseaton This code should not be changed.
Probably a rebase error.

chrisseaton commented:
I've removed the `Truffle::Support` module, done a lot of other cleanup that this exposed, and also moved `lib/rubysl/*` into `lib/truffle` and moved everything under the `RubySL` module into `Truffle`.

chrisseaton commented:
Leaving to @eregon to review again.

eregon commented on src/main/ruby/core/file.rb:
```diff
@@ -312,26 +312,7 @@ def chown(owner, group)
-    end
-
-    paths.size
+    raise NotImplementedError, 'lchown not implemented on this platform'
```
I would rather remove the first line, since the rest of the logic should work just fine.

chrisseaton replied:
It's not tested - it's dead code. But ok I'll look at running the super user tests in Docker.

eregon commented on lib/truffle/date.rb:
```diff
@@ -277,11 +277,11 @@ def <=> (other)
-	  return l <=> r
-	rescue NoMethodError
-	end
+	       begin
```
This looks wrong. Are there tabs in the file?

chrisseaton replied:
There are.

chrisseaton commented:
I can squash all the commits that replace `Rubinius` with `Truffle` as a first step, but after that I'm still in favour of merging the Rubinius code into ours and bring it into line with our style configuration, but will discuss that offline.

chrisseaton commented:
It was more difficult than I thought to keep more commits, as they were fixes to things that only broken when I checked more tests, so I've squashed more of them.

eregon commented on tool/generate-native-config.rb:
```diff
@@ -35,7 +35,7 @@
 # and its required files, but greatly simplified.
 
-PREFIX = 'rbx.platform'
+PREFIX = 'platform'
```
I think we can even simplify to an empty `PREFIX`, but should be another PR.
